pkg/action/drutils.go

```diff
@@ -162,3 +162,62 @@ func (ac *ActionController) remoteClusterDomainUpdate(activate bool, action *sto
 	}
 	return nil
 }
+
+func (ac *ActionController) updateApplicationActivatedInRelevantMigrationSchedules(action *storkv1.Action, config *rest.Config, namespaces []string, migrationNamespaces []string, value bool) {
+	// This is best effort only. In case we encounter an error we just log events and logs. Action will not be failed.
+
+	// Only subset of namespaces which are migrated by the reference migrationSchedule are activated / deactivated
+	_, activationNamespaces, _ := utils.IsSubList(namespaces, migrationNamespaces)
+	// relevant migrationSchedules are ones which are involved in migration of resources from at least one of the namespaces being activated.
+	// such migrationSchedules can reside in any of the activationNamespaces or in the adminNamespace
+	adminNs := utils.GetAdminNamespace()
+	migrationScheduleNamespaces := activationNamespaces
+	if !slices.Contains(activationNamespaces, adminNs) {
+		migrationScheduleNamespaces = append(migrationScheduleNamespaces, adminNs)
+	}
+
+	storkClient, err := storkops.NewForConfig(config)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to create the stork client to access cluster with the config %v", config.Host)
+		logEvents := ac.printFunc(action, "ApplicationActivatedStatus")
+		logEvents(msg, "err")
+		action.Status.Status = storkv1.ActionStatusFailed
+		ac.updateAction(action)
+		return
+	}
+
+	migrationSchedules := new(storkv1.MigrationScheduleList)
+	for _, ns := range migrationScheduleNamespaces {
+		migrSchedules, err := storkClient.ListMigrationSchedules(ns)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to fetch the list of migrationSchedules in the namespace %v", ns)
+			logEvents := ac.printFunc(action, "ApplicationActivatedStatus")
+			logEvents(msg, "err")
+			ac.updateAction(action)
+		}
+		migrationSchedules.Items = append(migrationSchedules.Items, migrSchedules.Items...)
+	}
+
+	for _, migrSched := range migrationSchedules.Items {
+		isMigrSchedRelevant, err := utils.DoesMigrationScheduleMigrateNamespaces(migrSched, activationNamespaces)
```

Comment:
Let's check if the migrationschedule has the `static copy`annotation.

Reply:
Makes sense since updating ApplicationActivated only makes sense in static copy migration schedule.
Will address